### DISCUSSION TITLE
Core functionality of Nuclei

### DISCUSSION THREAD

**Discussion by Prankster1028** [projectdiscovery/nuclei#707] [Q&A] [answered]
How does Nuclei send requests based on the templates? 
Further, what is the basic workflow? Want to know in order to understand the tool fully.

**Comment by ehsandeep** [comment, marked as answer]
@Prankster1028 Nuclei supports multiple protocols including HTTP/DNS/TCP/FILE and all the codes specific to protocols listed below:-

Protocols:- https://github.com/projectdiscovery/nuclei/tree/master/v2/pkg/protocols
Workflows:- https://github.com/projectdiscovery/nuclei/tree/master/v2/pkg/workflows
